backend/src/main/java/be/cytomine/appstore/services/TaskValidationService.java

```diff
@@ -0,0 +1,141 @@
+    }
+
+    public void validateImage(UploadTaskArchive task) throws ValidationException {
+        checkManifestJsonExists(task);
+    }
+
+    private void checkManifestJsonExists(UploadTaskArchive task) throws ValidationException {
+        try (
+            FileInputStream fis = new FileInputStream(task.getDockerImage());
+            BufferedInputStream bis = new BufferedInputStream(fis);
+            TarArchiveInputStream tais = new TarArchiveInputStream(bis)
+        ) {
+            TarArchiveEntry tarArchiveEntry;
+
+            while ((tarArchiveEntry = tais.getNextTarEntry()) != null) {
+                String name = tarArchiveEntry.getName();
+                if (name.equalsIgnoreCase("manifest.json")) {
+                    return;
+                }
+            }
+        } catch (IOException e) {
+            log.error("Failed to check for manifest.json in the Docker image", e);
+            AppStoreError error = ErrorBuilder.build(
+                ErrorCode.INTERNAL_DOCKER_IMAGE_EXTRACTION_FAILED
+            );
+            throw new ValidationException(error);
+        }
+
+        log.info("Validation error [manifest.json does not exist]");
+        AppStoreError error = ErrorBuilder.build(ErrorCode.INTERNAL_DOCKER_IMAGE_MANIFEST_MISSING);
+        throw new ValidationException(error);
+    }
+
+    public void validateDescriptorFile(UploadTaskArchive archive) throws ValidationException {
+        Set<ValidationMessage> errors = getDescriptorJsonSchemaV7()
+            .validate(archive.getDescriptorFileAsJson());
+        // prepare error list just in case
+        List<AppStoreError> multipleErrors = new ArrayList<>();
+        for (ValidationMessage message : errors) {
+            AppStoreError error = buildErrorFromValidationMessage(message);
+            multipleErrors.add(error);
+        }
+
+        if (!multipleErrors.isEmpty()) {
+            AppStoreError error = ErrorBuilder.buildSchemaValidationError(multipleErrors);
+            throw new ValidationException(error);
+        }
+    }
+
+    @NotNull
+    private static AppStoreError buildErrorFromValidationMessage(ValidationMessage message) {
+        String parameterPathInSchema = message
+            .getMessage()
+            .substring(0, message.getMessage().indexOf(':'))
+            .replace("$.", "");
+        ParameterError parameterError = new ParameterError(parameterPathInSchema);
+        String formattedMessage = message
+            .getMessage()
+            .replace("$.", "")
+            .replace(":", "")
+            .replace(".", " ");
+        AppStoreError error = ErrorBuilder.buildWithMessage(
+            ErrorCode.INTERNAL_PARAMETER_SCHEMA_VALIDATION_ERROR,
+            formattedMessage,
+            parameterError
+        );
+        return error;
+    }
+
+    private static JsonSchema getDescriptorJsonSchemaV7() throws ValidationException {
+        ClassPathResource resource = new ClassPathResource("/schemas/tasks/task.v0.json");
```

Comment:
To avoid duplicating the json schema, which can lead to error in the future, instead use `UrlResource` and point to the app engine schema directly. For instance, 
`UrlResource('https://github.com/cytomine/cytomine/blob/main/app-engine/src/main/resources/schemas/tasks/task.v0.json')`